src/emailpro/emailpro.service.js

```diff
@@ -212,22 +212,101 @@ angular
       /**
        * Return the last 2 days task list for the selected exchange
        */
-      getTasks(serviceName, pageSize, offset) {
+      getTasks(serviceName, pageSize, offset, domainName) {
         return this.gettingIsServiceMXPlan()
-          .then(isMXPlan => this.OvhHttp
-            .get('/sws/emailpro/{exchange}/tasks', {
-              rootPath: '2api',
-              urlParams: {
-                exchange: serviceName,
-              },
-              params: {
-                count: pageSize || 10,
-                offset: offset || 0,
-                isMXPlan,
-              },
-            }));
+          .then(isMXPlan => {
+            if(isMXPlan) {
+              return this.$q.all([
+                this.getRedirectionTasks(domainName),
+                this.getMailingListTasks(domainName),
+                this.getMxTasks(serviceName),
+              ]);
```

Comment:
No catch case ? (And also for all promise method)

Reply:
Having a single catch in controller. Since its `$q.all` any failure will return in `q` to return an error and displaying a common error message